### ISSUE TITLE
TypeScript definitions

### ISSUE THREAD

**Issue by ulfryk** [cwmyers/monet.js#37]
Is there any possibility that you will add TypeScript definitions of this great library to [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) or to this repository ?

I would feel happy to help with that :)

**Comment by dumpstate** [comment]
+1

**Comment by cwmyers** [comment]
Great idea

**Comment by michalma** [comment]
+1

**Comment by cwmyers** [comment]
@ulfryk If you were willing to get this kick-started that would be amazing.

**Comment by ulfryk** [comment]
Some work is already done -> https://gist.github.com/ulfryk/0687f6b6d454c84dcff7

**Comment by ulfryk** [comment]
I have quite huge problem with typing List. Part Below is **NOT** a valid TypeScript…

```
class List<T> {
    …
    sequence<V, T extends Monad<V>>(m: IMonadStatic): Monad<List<V>>;
    sequenceMaybe<V, T extends Maybe<V>>(): Maybe<List<V>>;
    sequenceEither<E, V, T extends Either<E, V>>(): Either<E, List<V>>;
    sequenceValidation<E, V, T extends Validation<E, V>>(): Validation<List<E>, List<V>>;
    sequenceIO<V, T extends IO<V>>(): IO<List<V>>;
    sequenceReader<V, T extends Reader<V>>(): Reader<List<V>>;
    …
}
```



I can't find proper way to express possibility that in **`List<T>`** the **`T`** is **`Monad<V>`**…

Maybe.Some ideas ?

**Comment by ulfryk** [comment]
@cwmyers - you can see more progress here -> https://github.com/evojam/monet.js

PR's are coming soon!

**Comment by ulfryk** [comment]
PR is ready: #48

**Comment by ulfryk** [comment]
Ready! Released with patch `0.8.9` 👯
